src/esp/gfx/GenericShader.h

```diff
@@ -54,10 +54,8 @@ class GenericShader : public Magnum::GL::AbstractShaderProgram {
   enum : uint8_t {
```

Comment:
Surprised. I thought such enum definition would require C++17, while we were using C++14 in this project.

@mosra: 
Are these outputs allso required to be "mutually exclusive" for each shader?

For textures, I introduced a new header file called `TextureBindingPointIndex.h` in #132 to define enumerators (binding points) so that texture units would not conflict. I would like to move the "textureLayer" defined in GenericShader.cpp to this file. Any comments?

Reply:
I think this enum syntax was available in c++11:

https://en.cppreference.com/w/cpp/language/enum

Reply:
Yes, C++11 added two new things to `enum`s -- strong enums (`enum class`) and typed enums (`enum: type`). Those two are completely orthogonal features, one can be used without the other.

Texture binding points -- see my other comment at https://github.com/facebookresearch/habitat-sim/pull/151#discussion_r314031297